src/api/app/controllers/webui/feeds_controller.rb

```diff
@@ -38,15 +38,14 @@ def notifications
 
   private
 
-  def starting_at(date)
-    Time.zone.parse(date)
-  rescue StandardError
-    7.days.ago
-  end
-
-  def ending_at(date)
-    Time.zone.parse(date)
-  rescue StandardError
-    nil
+  def set_timerange
+    start = params.fetch(:starting_at, 7.days.ago.to_s)
+    @start = Time.zone.parse(start)
+    finish = params.fetch(:ending_at, nil)
```

Comment:
Using `fetch` isn't needed if you set the default value to `nil`:

```suggestion
    finish = params[:ending_at]
```

Reply:
https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-fetch

Reply:
but as you wish :)